### PR TITLE
Add Extra Hosts docs

### PR DESCRIPTION
**Pull request by wolfgangbecker**
## [FX-4202](https://zombie.atlassian.net/browse/FX-4202)

❗️IMPORTANT: Do not merge yet. 
We'll deploy this once the feature goes live.

### Added

| Page | Deploy Preview | Comment |
| ----- | ----- | ----- |
| Extra Hosts section | [Link](https://deploy-preview-504--cobalt-docs.netlify.app/platform-deep-dive/scans/extra_hosts/) | Explains what extra hosts are for and how to configure them |

## Preview This Change

To see how this change looks in production, scroll down to **Deploy Preview**. Select the link that looks like `https://deploy-preview-<num>--cobalt-docs.netlify.app/`

## Variables

Help us support a “Write once, publish everywhere” single source of truth. If you see a line that looks like:

`{{% asset-categories %}}`

You’ve found a [shortcode](https://gohugo.io/content-management/shortcodes/) that we include in multiple documents.

You’ll find the content of the shortcode in the following directory:

https://github.com/cobalthq/cobalt-product-public-docs/tree/main/layouts/shortcodes

That shortcode has the same base name as what you see in the PR, such as `asset-categories.html`.

## Checklist for PR Author

[ ] Did you check for broken links and alt text?

Be sure to check for broken links and Alt Text issues. We have a partially automated process,
as described in this section of our repository README:
[Test Links and Alt Attributes](https://github.com/cobalthq/cobalt-product-public-docs/blob/main/README.md#test-links-and-alt-attributes).

[FX-4202]: https://zombie.atlassian.net/browse/FX-4202?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: content/en/Platform Deep Dive/Scans/extra_hosts.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+title: "Extra Hosts"
+linkTitle: "Extra Hosts"
+weight: 10
+description: >
+  Add extra hosts to include further domains in Web Application scans.
+---
+
+## Description
+
+![Extra Hosts page](/deepdive/scans/extra-hosts/ExtraHosts_List.png "Extra Hosts page")
```
Comment (mahmoudadly-cobalt) [inline content/en/Platform Deep Dive/Scans/extra_hosts.md:11]: You can now use the new image styles 🎉 
```suggestion
{{% image src="/deepdive/scans/extra-hosts/ExtraHosts_List.png" alt="Extra Hosts page" %}}
```

File: content/en/Platform Deep Dive/Scans/extra_hosts.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+An extra host serves the purpose of including non-origin domains in the scope of a Web target's scan. 
+Oftentimes, a web application will send requests to a domain that's different from where it was loaded from. By default, those interactions won't be taken into account when scanning for vulnerabilities. However, if these domains are specified as extra hosts, they will be included by the scanner.
+
+## Configuration
+
+![Extra Hosts form](/deepdive/scans/extra-hosts/ExtraHosts_Create.png "Extra Hosts form")
```
Comment (mahmoudadly-cobalt) [inline content/en/Platform Deep Dive/Scans/extra_hosts.md:18]:
```suggestion
{{% image src="/deepdive/scans/extra-hosts/ExtraHosts_Create.png" alt="Extra Hosts form" %}}
```

**Review by mtsuji-cobalt — COMMENTED** [review]
in terms of hierarchy, should we move this to below Targets (or Authentication)?

Screenshot 2024-06-27 at 08 31 54

**Review by eugenerevzin — APPROVED** [review]
This looks great!  One minor change - could we move the "Extra Hosts" page down in the left nav (maybe below Sequence Recorder?)?

**Review by mahmoudadly-cobalt — APPROVED** [review]

File: content/en/Platform Deep Dive/Scans/extra_hosts.md
Comment (mahmoudadly-cobalt) [inline content/en/Platform Deep Dive/Scans/extra_hosts.md:1]: I've just noticed one tiny thing: the file name should  probably be `extra-hosts.md` to have a consistent URL with `-` separators.

**Review by mahmoudadly-cobalt — COMMENTED** [review]
Not a blocker.

File: content/en/Platform Deep Dive/Scans/extra_hosts.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+description: >
+  Add extra hosts to include further domains in Web Application scans.
+---
+
+## Description
+
+{{% image src="/deepdive/scans/extra-hosts/ExtraHosts_List.png" alt="Extra Hosts page" %}}
+
+An extra host serves the purpose of including non-origin domains in the scope of a Web target's scan. 
+Oftentimes, a web application will send requests to a domain that's different from where it was loaded from. By default, those interactions won't be taken into account when scanning for vulnerabilities. However, if these domains are specified as extra hosts, they will be included by the scanner.
```
Comment (davidgm0) [inline content/en/Platform Deep Dive/Scans/extra_hosts.md:14]: I think an example could help clarifying what is a "host" (specially in regards of subdomains) and in which cases is useful. In Probely they mention SPAs and API's for instance https://help.probely.com/en/articles/1975501-what-are-the-extra-hosts-in-scope-for

Comment (wolfgangbecker) [reply content/en/Platform Deep Dive/Scans/extra_hosts.md:14]: Good one, will add the example mentioned in that link.

File: content/en/Platform Deep Dive/Scans/extra_hosts.md
Comment (wolfgangbecker) [reply content/en/Platform Deep Dive/Scans/extra_hosts.md:1]: Makes sense, will also rename `target_auth.md` for consistency.

File: content/en/Platform Deep Dive/Scans/extra-hosts.md
Comment (mahmoudadly-cobalt) [inline content/en/Platform Deep Dive/Scans/extra-hosts.md:1]: Also this renaming will need an update in https://github.com/cobalthq/cobalt-app-web/pull/6316/files

Comment (wolfgangbecker) [reply content/en/Platform Deep Dive/Scans/extra-hosts.md:1]: Good catch!

**Review by mahmoudadly-cobalt — APPROVED** [review]

**Review by mohitkyadav — APPROVED** [review]

**Review by mohitkyadav — APPROVED** [review]

**Pull request merged**